### PR TITLE
Fix python handling

### PR DESCRIPTION
**Pull request by cournape**
First step to support repack for python 3.

The second step is updating enstaller to support those new arguments

@sjagoe

### CODE REVIEW

File: okonomiyaki/file_formats/setuptools_egg.py [hunk 1]
```diff
@@ -1,8 +1,13 @@
 import os.path
 import re
+try:
+    import sysconfig
+except ImportError:  # Pyton 2.6 support
```
Comment (sjagoe) [inline okonomiyaki/file_formats/setuptools_egg.py]: typo: `python`

File: okonomiyaki/file_formats/setuptools_egg.py [hunk 2]
```diff
@@ -74,37 +79,84 @@ def _get_default_setuptools_abi(platform_string, pyver):
             raise ValueError(msg)
 
 
+class _UnspecifiedArg(object):
+    pass
+_UNSPECIFIED = _UnspecifiedArg()
```
Comment (sjagoe) [inline okonomiyaki/file_formats/setuptools_egg.py]: I think you probably don't need a new class for this, just an instance of `object`.

File: okonomiyaki/file_formats/setuptools_egg.py [hunk 3]
```diff
@@ -74,37 +79,84 @@ def _get_default_setuptools_abi(platform_string, pyver):
...
+def _guess_abi(platform):
+    if platform is None:
+        return None
+
+    if sysconfig is None:
+        soabi = None
+    else:
+        try:
+            soabi = sysconfig.get_config_var('SOABI')
+        except IOError as e:  # Issue #1074
```
Comment (sjagoe) [inline okonomiyaki/file_formats/setuptools_egg.py]: Issue against what project? Python?

File: okonomiyaki/file_formats/setuptools_egg.py [hunk 4]
```diff
@@ -74,37 +79,84 @@ def _get_default_setuptools_abi(platform_string, pyver):
...
+            warnings.warn("{0}".format(e), RuntimeWarning)
+            soabi = None
+
+    if soabi and soabi.startswith('cpython-'):
+        return 'cp' + soabi.split('-', 1)[-1]
+    else:
+        msg = ("Could not guess ABI, you need to specify the abi_tag "
+               "argument to from_egg, e.g. 'cp34m' for Enthought "
+               "cpython 3.4 runtimes")
```
Comment (sjagoe) [inline okonomiyaki/file_formats/setuptools_egg.py]: `CPython`

File: okonomiyaki/file_formats/setuptools_egg.py [hunk 5]
```diff
@@ -74,37 +79,84 @@ def _get_default_setuptools_abi(platform_string, pyver):
...
+            msg = ("Platform-specific egg detected: you need to specify a "
+                   "platform argument that is not None to from_egg")
+            raise OkonomiyakiError(msg)
 
-        abi_tag = _get_default_setuptools_abi(platform, pyver)
+        if python_tag is _UNSPECIFIED:
+            python_tag = _guess_python_tag(pyver)
 
-        if platform is not None:
-            platform_tag = _normalize_setuptools_platform_string(platform)
```
Comment (sjagoe) [inline okonomiyaki/file_formats/setuptools_egg.py]: Is `_normalize_setuptools_platform_string` still used anywhere?

File: okonomiyaki/file_formats/setuptools_egg.py [hunk 6]
```diff
@@ -74,37 +79,84 @@ def _get_default_setuptools_abi(platform_string, pyver):
...
-
-    def __init__(self, name, version, python_tag, abi_tag, platform_tag,
-                 pkg_info):
+        return cls(name, version, platform, python_tag, abi_tag, pkg_info)
+
+    def __init__(self, name, version, platform, python_tag, abi_tag, pkg_info):
+        """
+        Parameters
+        ----------
+        name
```
Comment (sjagoe) [inline okonomiyaki/file_formats/setuptools_egg.py]: Incomplete docstring

File: okonomiyaki/file_formats/setuptools_egg.py (hunk 2 above)
Comment (cournape) [inline okonomiyaki/file_formats/setuptools_egg.py]: Hm, for some reason I thought you could not create instances of object directly !

File: okonomiyaki/file_formats/setuptools_egg.py (hunk 5 above)
Comment (cournape) [inline okonomiyaki/file_formats/setuptools_egg.py]: no, removed

**Pull request merged**